File: js/ledger.js
const {Web3,HttpProvider} = require('web3');
const fs=require('fs');
const { resolve } = require('path');

blockchain="http://127.0.0.1:7545"

const httpprovider=new HttpProvider(blockchain);
const web3=new Web3(httpprovider);

artifact_demo="../build/contracts/demo.json";

function assign(wallet,b) {
    var data=fs.readFileSync(artifact_demo,"utf-8");
    data=JSON.parse(data);
    contract_abi=data['abi'];
    contract_address=data['networks']['5777']['address'];

    contract=new web3.eth.Contract(contract_abi,contract_address);
    console.log('Contract Selected');
    try {
        
        contract.methods.assign(b).send({
            from: wallet,
            gas: 1000000,
            gasPrice: 1000000000000
        }).then((data)=>{
            console.log(data);
        })   
    } catch (error) {
        console.log(error);
        return 'Transaction Failed'
    }

    return 'Transaction Done    '

}

async function print(wallet) {
    
    var data=fs.readFileSync(artifact_demo,"utf-8");
    data=JSON.parse(data);
    contract_abi=data['abi'];
    contract_address=data['networks']['5777']['address'];

    contract=new web3.eth.Contract(contract_abi,contract_address);
    console.log('Contract Selected');

    try {
        response=await contract.methods.print().call();
        console.log(response);
        return response;
    } catch (error) {
        console.log(error);
        return 'something went wrong'
    }
}

module.exports={
    assign,
    print
}

File: src/ledger.py
from web3 import Web3,HTTPProvider
import json

blockchain="http://127.0.0.1:7545"

def connect_with_demo(wallet):
    # Step - 1: Connect with Blockchain Server
    web3=Web3(HTTPProvider(blockchain))
    print('Connected with Ganache')

    # Step - 2: ABI, Contract Address
    with open('../build/contracts/demo.json') as f:
        artifact_json=json.load(f)
        contract_abi=artifact_json['abi']
        contract_address=artifact_json['networks']['5777']['address']
    
    # Step - 3: Connect with Contract
    web3.eth.defaultAccount=wallet
    contract=web3.eth.contract(
        abi=contract_abi,
        address=contract_address
    )
    return (web3,contract)

# Sub - Functions
def assign(wallet,b):
    try:
        web3,contract=connect_with_demo(wallet)
        tx_hash=contract.functions.assign(b).transact()
        web3.eth.waitForTransactionReceipt(tx_hash)
        return 'transaction done'
    except:
        return 'transaction failed'

def prin(wallet):
    try:
        web3,contract=connect_with_demo(wallet)
        data=contract.functions.print().call()
        return data
    except:
        return 'transaction failed'